tests/qiskit_qir/test_qiskit_to_qir.py

```
@@ -0,0 +1,224 @@
+        module = qiskit_to_qir(bell_circuit, record_output=False)
+        ir = str(module)
+        assert "__quantum__rt__result_record_output" not in ir
+        assert "__quantum__rt__array_record_output" not in ir
+
+
+class TestQiskitToQirErrors:
+    """Tests for error handling in qiskit_to_qir."""
+
+    def test_invalid_input_type(self):
+        """Test that TypeError is raised for non-QuantumCircuit input."""
+        with pytest.raises(TypeError, match="must be of type qiskit.QuantumCircuit"):
```

Ruff (0.15.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)